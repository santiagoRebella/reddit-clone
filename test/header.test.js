const React = require('react');
const Header = require('components/header');
const renderer = require('react-test-renderer');

jest.mock('react-router-dom', () => ({
  NavLink: 'NavLink'
}))

test('Header is rendered correctly', () => {
  const component = renderer.create(
    <Header view="/" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const componentPost = renderer.create(
    <Header view="/something" />
  );

  const postTree = componentPost.toJSON();
  expect(postTree).toMatchSnapshot();
});
