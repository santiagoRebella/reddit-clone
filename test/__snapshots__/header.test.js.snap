// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is rendered correctly 1`] = `
<header
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 50,
    }
  }
>
  <img
    src="assets/logo-wide.svg"
    style={
      Object {
        "height": 40,
        "marginRight": 5,
      }
    }
  />
  <nav
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <NavLink
      style={
        Object {
          "fontWeight": 600,
        }
      }
      to="/"
    >
      Subreddits
    </NavLink>
  </nav>
</header>
`;

exports[`Header is rendered correctly 2`] = `
<header
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 50,
    }
  }
>
  <img
    src="assets/logo-wide.svg"
    style={
      Object {
        "height": 40,
        "marginRight": 5,
      }
    }
  />
  <nav
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <NavLink
      style={
        Object {
          "fontWeight": 300,
        }
      }
      to="/"
    >
      Subreddits
    </NavLink>
    <NavLink
      style={
        Object {
          "fontWeight": 600,
        }
      }
      to="/something"
    >
      Â &gt; 
      /something
    </NavLink>
  </nav>
</header>
`;
